Hitting a RecursionError with gevent

I'm hitting the following `RecursionError` when a part of an application ([pywb](https://pypi.org/project/pywb/) delivering static files) tries to respond (the more complex functionalities work. :-)). i think that exception should be handled / propagated since i can't deduce where something went wrong in the app from this output.

```
Traceback (most recent call last):
  File "/usr/local/share/virtualenv/pywb/lib/python3.6/site-packages/gunicorn/workers/base_async.py", line 111, in handle_request
    resp.write_file(respiter)
  File "/usr/local/share/virtualenv/pywb/lib/python3.6/site-packages/gunicorn/http/wsgi.py", line 392, in write_file
    if not self.sendfile(respiter):
  File "/usr/local/share/virtualenv/pywb/lib/python3.6/site-packages/gunicorn/http/wsgi.py", line 382, in sendfile
    sent += os.sendfile(sockno, fileno, offset + sent, count)
  File "/usr/local/share/virtualenv/pywb/lib/python3.6/site-packages/gunicorn/workers/ggevent.py", line 37, in _gevent_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  File "/usr/local/share/virtualenv/pywb/lib/python3.6/site-packages/gunicorn/workers/ggevent.py", line 37, in _gevent_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  File "/usr/local/share/virtualenv/pywb/lib/python3.6/site-packages/gunicorn/workers/ggevent.py", line 37, in _gevent_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  [Previous line repeated 923 more times]
RecursionError: maximum recursion depth exceeded
```

I am also seeing this after updating from Gunicorn 19.9.0 to 20.0.0 (with gevent 1.4.0, Django 2.2.6, Python 3.7.5, Ubuntu 19.10):

```
[2019-11-11 09:16:23 +0000] [11635] [ERROR] Error handling request
Traceback (most recent call last):
  File "/data-hub-api/lib/python3.7/site-packages/gunicorn/workers/base_async.py", line 111, in handle_request
    resp.write_file(respiter)
  File "/data-hub-api/lib/python3.7/site-packages/gunicorn/http/wsgi.py", line 397, in write_file
    if not self.sendfile(respiter):
  File "/data-hub-api/lib/python3.7/site-packages/gunicorn/http/wsgi.py", line 387, in sendfile
    sent += os.sendfile(sockno, fileno, offset + sent, count)
  File "/data-hub-api/lib/python3.7/site-packages/gunicorn/workers/ggevent.py", line 36, in _gevent_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  File "/data-hub-api/lib/python3.7/site-packages/gunicorn/workers/ggevent.py", line 36, in _gevent_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  File "/data-hub-api/lib/python3.7/site-packages/gunicorn/workers/ggevent.py", line 36, in _gevent_sendfile
    return os.sendfile(fdout, fdin, offset, nbytes)
  [Previous line repeated 936 more times]
RecursionError: maximum recursion depth exceeded
```

Is there any known solution?

--- a/gunicorn/workers/geventlet.py
+++ b/gunicorn/workers/geventlet.py
@@ -26,10 +26,10 @@ import greenlet
 from gunicorn.workers.base_async import AsyncWorker
 
 
-def _eventlet_sendfile(fdout, fdin, offset, nbytes):
+def _eventlet_sendfile(fdout, fdin, offset, nbytes, _os_sendfile=os.sendfile):
     while True:
         try:
-            return os.sendfile(fdout, fdin, offset, nbytes)
+            return _os_sendfile(fdout, fdin, offset, nbytes)
         except OSError as e:
             if e.args[0] == errno.EAGAIN:
                 trampoline(fdout, write=True)

--- a/gunicorn/workers/ggevent.py
+++ b/gunicorn/workers/ggevent.py
@@ -30,10 +30,10 @@ from gunicorn.workers.base_async import AsyncWorker
 VERSION = "gevent/%s gunicorn/%s" % (gevent.__version__, gunicorn.__version__)
 
 
-def _gevent_sendfile(fdout, fdin, offset, nbytes):
+def _gevent_sendfile(fdout, fdin, offset, nbytes, _os_sendfile=os.sendfile):
     while True:
         try:
-            return os.sendfile(fdout, fdin, offset, nbytes)
+            return _os_sendfile(fdout, fdin, offset, nbytes)
         except OSError as e:
             if e.args[0] == errno.EAGAIN:
                 socket.wait_write(fdout)

